examples/karpenter-fargate/provisioners/sample_deployment.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: apps/v1
```

Comment:
Can we use the existing [app](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/modules/kubernetes-addons/app-2048) instead?

Reply:
I thought about using it at first, but finally decided to use a deployment with `nodeSelector`, `topologySpreadConstraints`, and `podAntiAffinity` to demonstrate how Karpenter handles these types of constraints. I explain it in the [README](https://github.com/micbegin/terraform-aws-eks-blueprints/blob/karpenter-fargate/examples/karpenter-fargate/README.md?plain=1#L157). I understand that this may look like code duplication, so I can replace it with a simpler deployment using pause pods like this [inflate example](https://www.eksworkshop.com/beginner/085_scaling_karpenter/automatic_node_provisioning/) with the same constraints if you prefer.

Reply:
@askulkarni2 I just realized that I may not have been clear in what I'm trying to achieve with this sample deployment. I want to deploy one application to demonstrate two things:

1. Deploy workloads with contraints to see how Karpenter will manage them
2. Deploy a web application accessible through an ALB provisioned by the AWS Load Balancer Controller running on Fargate

I thought that using the same application to demonstrate these two points was a good idea so that's why I didn't use the [app-2048](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/modules/kubernetes-addons/app-2048) addon.

Reply:
1. What are we trying to highlight or demonstrate?
2. Why is this beneficial to customers?
3. Is this specific to Karpenter on Fargate?

Reply:
We are trying to demonstrate how to create a cluster using Fargate profiles in which to run Karpenter and other add-ons. More specifically how to implement the [Run the Karpenter controller on EKS Fargate](https://aws.github.io/aws-eks-best-practices/karpenter/#run-the-karpenter-controller-on-eks-fargate-or-on-a-worker-node-that-belongs-to-a-node-group) best practice and provide a demo application so customers can test this use case using the EKS blueprints.

This example also uses spot instances, SQS for the interruption queue and an ALB deployed by the AWS Load Balancer Controller (running on Fargate) to access the demo app (running on worker nodes deployed by Karpenter).

The demo app deployment is also using `nodeSelector`, `topologySpreadConstraints`, and `podAntiAffinity` to demonstrate how Karpenter handles these types of constraints.